src/sqlfluff/utils/reflow/depthmap.py

```diff
@@ -33,7 +33,7 @@ class DepthInfo:
     # This is a convenience cache to speed up operations.
     stack_hash_set: FrozenSet[int]
     stack_class_types: Tuple[FrozenSet[str], ...]
-    stack_positions: Tuple[str, ...]
+    stack_positions: Dict[int, Tuple[int, int, str]]
```

Comment:
Think it's worth introducing a `NamedTuple` or `dataclass` here in place of the anonymous tuple structure, `Tuple[int, int, str]`?

Reply:
Yeah good idea.

Reply:
Done. Converted to frozen dataclass.